src/main.rs

```diff
@@ -18,34 +20,48 @@ fn main() {
     loop {
         println!("Please enter a frontend problem id, or \"random\" to generate a random one.");
         let mut is_random = false;
+        let mut is_solving = false;
         let mut id: u32 = 0;
         let mut id_arg = String::new();
         io::stdin()
             .read_line(&mut id_arg)
             .expect("Failed to read line");
         let id_arg = id_arg.trim();
-        match id_arg {
-            "random" => {
-                println!("You select random mode.");
-                id = generate_random_id(&solved_ids);
-                is_random = true;
-                println!("Generate random problem: {}", &id);
-            }
-            _ => {
-                id = id_arg
-                    .parse::<u32>()
-                    .unwrap_or_else(|_| panic!("not a number: {}", id_arg));
-                if solved_ids.contains(&id) {
-                    println!(
-                        "The problem you chose is invalid (the problem may have been solved \
-                         or may have no rust version)."
-                    );
-                    continue;
-                }
+
+        let random_pattern = Regex::new(r"^random$").unwrap();
+        let solving_pattern = Regex::new(r"^solve (\d+)$").unwrap();
+
+        if random_pattern.is_match(id_arg) {
+            println!("You select random mode.");
+            id = generate_random_id(&solved_ids);
+            is_random = true;
+            println!("Generate random problem: {}", &id);
+        } else if solving_pattern.is_match(id_arg) {
+            // solve a problem
+            // move it from problem/ to solution/
+            is_solving = true;
+            id = solving_pattern
+                .captures(id_arg)
+                .unwrap()
+                .get(1)
+                .unwrap()
+                .as_str()
+                .parse()
+                .unwrap();
+        } else {
+            id = id_arg
+                .parse::<u32>()
+                .unwrap_or_else(|_| panic!("not a number: {}", id_arg));
+            if solved_ids.contains(&id) {
+                println!(
+                    "The problem you chose is invalid (the problem may have been initialized \
+                     or may have no rust version)."
```

Comment:
the error message seems not match the condition that the problem has been solved.

Reply:
fixed in https://github.com/aylei/leetcode-rust/pull/18/commits/125c64c39b035d4cab2687d9bc2977749c8b0a59 https://github.com/aylei/leetcode-rust/pull/18/commits/76edb8c5fb172de61a1e308d279e9b877ad32c5f https://github.com/aylei/leetcode-rust/pull/18/commits/695f53a684264b19f0d2e94541fdb18cc1f37628